### PR TITLE
Add database migrations framework

### PR DESCRIPTION
**Pull request by nwalters512**
This is inspired based on the system for database migrations that I wrote for PrairieLearn. Details:

- Each change to the database is expressed in a numbered file: `migrations/###_my_migration_name.sql`.
- When applied in order, these migrations should result in a database matching that which would be created by `sequelize.sync({ force: true })`.
- The database maintains a record of which migrations have been applied to it in a special `migrations` table.
- When the app is starting, it will run any unapplied migrations on the table.

CI will automatically check that the database created by Sequelize and the database created by our migrations match, and will fail if they don't.

### CODE REVIEW

**Comment by shreyas208** [comment]
Looks good overall, but I'm unclear on what exactly the migration workflow would be:
- What does a developer need to do when making changes that require DB schema changes?
- What does the person deploying an update need to do once these changes have been merged?
It would also be good to have this documented if it's non-trivial.

**Comment by nwalters512** [comment]
:+1: for documenting this process, I'll add that before this is merged.

The idea here is that the Sequelize model definitions serve as the source of truth for what the database should look like; migrations simply track changes to the model definitions over time and allow the app to automatically "fast-forward" the database to what it should look like, given that Sequelize can't (and probably shouldn't) do that automatically.

The process for changing the DB schema has two steps:

- Modify the Sequelize model definitions in `models`.
- Add a `migrations/###_xxxxxxx.sql` file that will change the database in the corresponding way.

CI will ensure that the migrations leave the database in the expected state, and will provide a suggested query to bring the database into sync.

The person deploying simply runs the app.

- If the app is being deployed to a host with a database that already has a `migrations` table, only new migrations will be run.
- If the app is being deployed to a host without a `migrations` table, that table will be created and all migrations will be run, and the database will begin tracking which migrations have been run.

**Comment by shreyas208** [comment]
Ah okay, thanks for clarifying that; two things that come to mind:
- We need to make it very clear that devs committing schema-changing code must also write and commit a separate SQL script; not a huge concern at the moment, but if/when we have more contributors this could be a point of confusion if not made explicit.
- Is there a possibility with this model of merging/squashing migrations somehow? This isn't crucial, but DB schema changes are pretty expensive, so it would be nice if an instance that's being updated across several releases didn't have to incrementally apply each migration separately. Or maybe, just for fresh installs, there could be a way to specify the latest schema without starting at 0 and applying `n` scripts? Neither of these should be happening frequently though, so I'm also fine with keeping it totally incremental.

File: README.md [hunk 1]
```diff
@@ -24,14 +24,46 @@ To create the database for development:
...
 CREATE USER 'queue'@'localhost';
 GRANT ALL PRIVILEGES ON *.* TO 'queue'@'localhost';
 ```
 
-### Multiple users in dev mode
+If you want to run the test suite, it's important that the `queue` user be able to create and destroy databases. Keep that in mind if you decide to change the privileges of the `queue` user.
+
+## Database Migrations
+
+We use [Sequelize](http://docs.sequelizejs.com/) as an ORM to make interacting with our database easier. However, its build-in support for migrations is primitive and hard to automate, so Queue includes its own migration support.
```
Comment (shreyas208) [inline README.md]: "build" -> "built"

File: README.md [hunk 2]
```diff
@@ -24,14 +24,46 @@ To create the database for development:
...
+
+Migrations are tracked in a `migrations` table, which stores the index of the migration, its filename, and when it was applied.
+
+The following steps occur when the app starts up in order to perform migrations (these are implemented in `migrations/index.js`):
+
+1. The database is queried to find the index of the last migration that was applied.
+
+2. All `.sql` files from `migrations` are read, parsed, and sorted by index.
+
+3. Each one is applied to the database sequentially in order. When a migration is successfully applied, an entry is created in the `migrations` table so that that migration won't be applied again.
```
Comment (shreyas208) [inline README.md]: sequentially == in order, no? :)

Comment (nwalters512) [reply README.md]: True, I guess "in series" would be the better term. Was trying to emphasize that they happen one after the other 😛

**Comment by shreyas208** [comment]
Sounds good :) I agree, merging is not worth the pain if you're seeing times that low in PL.

As for fresh DBs, I like that option; I might make a separate issue for it. I don't think it's important enough to hold up this one since an empty DB is pretty trivial to do schema changes on.

Otherwise, docs look good! Should be clear to anyone. Just the two pedantic comments above.

**Comment by nwalters512** [comment]
Sounds good! Feel free to make that issue. Once @wadefagen signs off on this, I'll get it merged and we can start building more cool things.

**Comment by nwalters512** [comment]
Closing in favor of #56.

**Pull request closed without merging**